File: server/app/schemas/programs.py
"""프로그램 도메인 Pydantic 스키마."""

from datetime import datetime

from pydantic import BaseModel, Field


class ProgramRoutineItem(BaseModel):
    routine_id: str
    name: str
    gym_name: str | None = None
    order_index: int


class ProgramItem(BaseModel):
    program_id: str
    name: str
    description: str | None = None
    created_at: datetime
    routines: list[ProgramRoutineItem] = Field(default_factory=list)


class ProgramListData(BaseModel):
    items: list[ProgramItem]


class CreateProgramRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    description: str | None = None
    routine_ids: list[str] = Field(..., min_length=1)


class UpdateProgramRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    description: str | None = None


class DeleteProgramData(BaseModel):
    message: str

File: app/src/services/programs.ts
import { apiFetch } from './api';

export interface ProgramRoutineItem {
  routine_id: string;
  name: string;
  gym_name: string | null;
  order_index: number;
}

export interface ProgramItem {
  program_id: string;
  name: string;
  description: string | null;
  created_at: string;
  routines: ProgramRoutineItem[];
}

export interface ProgramListData {
  items: ProgramItem[];
}

export interface DeleteProgramData {
  program_id: string;
  message: string;
}

export function getProgramList(token: string): Promise<ProgramListData> {
  return apiFetch<ProgramListData>('/api/v1/programs', { token });
}

export function getProgram(token: string, program_id: string): Promise<ProgramItem> {
  return apiFetch<ProgramItem>(`/api/v1/programs/${program_id}`, { token });
}

export function createProgram(
  token: string,
  name: string,
  routine_ids: string[],
  description?: string,
): Promise<ProgramItem> {
  return apiFetch<ProgramItem>('/api/v1/programs', {
    method: 'POST',
    token,
    body: JSON.stringify({ name, routine_ids, description }),
  });
}

export function updateProgram(
  token: string,
  program_id: string,
  fields: { name?: string; description?: string },
): Promise<ProgramItem> {
  return apiFetch<ProgramItem>(`/api/v1/programs/${program_id}`, {
    method: 'PATCH',
    token,
    body: JSON.stringify(fields),
  });
}

export function deleteProgram(token: string, program_id: string): Promise<DeleteProgramData> {
  return apiFetch<DeleteProgramData>(`/api/v1/programs/${program_id}`, {
    method: 'DELETE',
    token,
  });
}
